src/module/item/abstract-effect/schema.ts

```diff
@@ -126,5 +126,68 @@ type DurationDataSchema = {
     expiry: fields.StringField<EffectExpiryType, EffectExpiryType, true, true>;
 };
 
+type EffectBadgeCounterSchema = {
+    type: fields.StringField<"counter", "counter", true, false>;
+    labels: fields.ArrayField<fields.StringField<string, string, true, false>, string[], string[], false, true, true>;
+    value: fields.NumberField<number, number, true, false, true>;
+    min: fields.NumberField<number, number, false, true>;
+    max: fields.NumberField<number, number, false, true>;
+    loop: fields.BooleanField<boolean, boolean, false, false>;
+};
+
+type EffectBadgeValueSchema = {
+    type: fields.StringField<"value", "value", true, false>;
+    labels: fields.ArrayField<fields.StringField<string, string, true, false>, string[], string[], false, true, true>;
+    value: fields.NumberField<number, number, true, false, true>;
+    reevaluate: fields.SchemaField<
+        EffectBadgeValueReevaluateSchema,
+        SourceFromSchema<EffectBadgeValueReevaluateSchema>,
+        ModelPropsFromSchema<EffectBadgeValueReevaluateSchema>,
+        true,
+        true
+    >;
+};
+
+type EffectBadgeValueReevaluateSchema = {
+    /** The type of event that reevaluation should occur */
+    event: fields.StringField<BadgeReevaluationEventType, BadgeReevaluationEventType, true, false>;
+    /** The formula of this badge when it was of a "formula" type */
+    formula: fields.StringField<string, string, true, false, true>;
+    /** The initial value of this badge */
+    initial: fields.NumberField<number, number, false, false>;
+};
+
+type EffectBadgeFormulaSchema = {
+    type: fields.StringField<"formula", "formula", true, false>;
+    labels: fields.ArrayField<fields.StringField<string, string, true, false>, string[], string[], false, true, true>;
+    value: fields.StringField<string, string, true, false, true>;
+    evaluate: fields.BooleanField<boolean, boolean, false, false, false>;
+    reevaluate: fields.StringField<BadgeReevaluationEventType, BadgeReevaluationEventType, true, true, true>;
+};
+
+type EffectBadgeCounterSource = Omit<SourceFromSchema<EffectBadgeCounterSchema>, "loop"> & {
+    loop?: boolean;
+};
+
+/** A static value, including the result of a formula badge */
+type EffectBadgeValueSource = SourceFromSchema<EffectBadgeValueSchema> & {
+    min?: never;
+    max?: never;
+};
+
+type EffectBadgeFormulaSource = Omit<SourceFromSchema<EffectBadgeFormulaSchema>, "evaluate"> & {
+    evaluate?: boolean;
+    min?: never;
+    max?: never;
+};
```

Comment:
Can these be interfaces instead of intersections?

Reply:
Seems to work fine. I mostly went with the flow for extending schemas, which do use intersections instead of inheritance.